best-first-cv/script.whizzml

```diff
@@ -408,55 +448,127 @@
          (log-info (str "Error deleting resource " id " ignored"))
          false)))
 
-
-
 ;-----------------------------------------------------------------
 
 
-
+;; select-features
+;;
 ;; Do best-first feature selection.  Given a dataset and a target
 ;; number of features iteratively construct models for each feature,
 ;; evaluate them, and add the feature corresponding to the best
 ;; cv to the running set of features.  Stop when you reach the
-;; target number, or you run out of features.
+;; target number, you run out of features, or if you don't have enough
+;; performance improvement during some iterations
+;;
+;; Inputs:
+;;  dataset-id: (dataset-id) Dataset ID
+;;  nfeatures: (number) Number of features that we want to obtain
+;;  objective-id: (resource-id) ID of the objective field
+;;  options: (map) Options to use in model/ensemble
+;;  k-folds: (number) Number of folds
+;;  pre-selected: (list) List of IDs of pre-selected fields
+;;  early-stop-performance: (number) (%) Minimum required performance improvement
+;;  max-bad-performance: (number) (%) Maximum number of iterations with bad
+;;                                    performance, as percentage of the number
+;;                                    of initial features
+;;
+;; Output: (map) Response with selected fields names and iterations info
+;;
 (define (select-features dataset-id
                          nfeatures
                          objective-id
                          options
                          k-folds
-                         pre-selected)
+                         pre-selected
+                         early-stop-performance
+                         max-bad-pecentage)
   (let (obj-id (get-objective dataset-id objective-id)
-        k-fold-datasets (create-k-folds dataset-id k-folds)
+	k-fold-datasets (create-k-folds dataset-id k-folds)
```

Comment:
this should be aligned with obj-id

Reply:
It is strange, in the actual file, the indentation seems correct. Please, check new commit.